File: dp_primer/dp_primer_lis_continuous_boss.cs
// 実行時間 90ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int n = int.Parse( ReadLine() );
        byte[] a = new byte[n];
        int[] p = new int[n];
        int o = 1;

        foreach( int i in Range( 0, n ) ) {
            a[i] = byte.Parse( ReadLine() );
        }
        p[0] = 1;
        foreach( int i in Range( 1, n - 1 ) ) {
            p[i] = a[i] <= a[i - 1] ? p[i - 1] + 1 : 1;
            o = System.Math.Max( o, p[i] );
        }
        WriteLine( o );
        return;
    }
}

File: dp_primer/dp_primer_lis_continuous_boss.py
# 実行時間 120ms
import sys

rl = sys.stdin.readline
n = int(rl())
o = 1
a = []
p = [1]

for _ in [0] * n:
    a.append(int(rl()))
for i in range(1, n):
    t = p[i - 1] + 1 if a[i] <= a[i - 1] else 1

    p.append(t)
    if o < t:
        o = t
print(o)
